<launch>

<!-- corridors  nsh  smith-->
    <arg name="data_path" default = "corridors" />
    <arg name="cloud_name"  default="$(find map_fusion)/config/realsense/$(arg data_path)/pt.ply"/>
    <arg name="config_file" default="$(find map_fusion)/config/realsense/realsense.yaml" />
    <arg name="line3d_file" default="$(find map_fusion)/config/realsense/$(arg data_path)/line_3d.txt" />

    <node name="tracking_node" pkg="map_fusion" type="window_tracking_node" output="screen">
        <param name="config_file" type="string" value="$(arg config_file)" />
        <param name="lines_map" type="string" value="$(arg line3d_file)" />
        <!--remap from="~estimated_odometry" to="/vins_estimator/odometry" /> -->
    </node>
    
    <node name="realsense_d435i_publisher" pkg="initialization_publisher" type="realsense_d435i_publisher" output="screen">
        <param name="cloud_name" type="string" value="$(arg cloud_name)" />
        <param name="config_file" type="string" value="$(arg config_file)" />
        <remap from="~estimated_odometry" to="/vins_estimator/odometry" />
    </node> 
    
    <arg name="image_topic" default = "/camera/infra1/image_rect_raw" />
    <node pkg="afm" name="ros_test" type="ros_test.py" output="screen" cwd="node">
        <param name="config_file" type="string" value="$(find afm)/scripts/experiments/afm_unet_rs.yaml" />
        <param name="image"   type="string"  value="$(find afm)/scripts/data/EuRoc/images/img_0.jpg"/>
        <param name="gpu"     value="0" />
        <remap from="/cam0/image_raw" to="$(arg image_topic)" /> 
    </node> 
    
</launch>
